### PR TITLE
fix: Adding a defer func to connecttoTelemetryservice() to prevent CNI fro…

### PR DESCRIPTION
**Pull request by AzureAhai**
Adding a defer func to connecttoTelemetryservice() 

**Reason for Change**:
to prevent CNI from stucking when the telemetry service is unable to start.

### CODE REVIEW

File: telemetry/telemetrybuffer.go [hunk 1]
```diff
@@ -328,32 +328,42 @@ func (tb *TelemetryBuffer) ConnectCNIToTelemetryService(telemetryNumRetries, tel
...
+	}
+	return nil
+}
+
+// This function is getting called from ConnectCNIToTelemetryService() in each attempt inside for loop
+// This function has been created to be able to add defer within the for loop
+func (tb *TelemetryBuffer) ConnectCNIToTelemetryServiceAttempt(telemetryNumRetries, telemetryWaitTimeInMilliseconds int, netPlugin *cni.Plugin, path string, args []string) error {
+	if err := tb.Connect(); err != nil {
+		log.Logf("Connection to telemetry socket failed: %v", err)
+		if runtime.GOOS == "windows" {
```
Comment (tamilmani1989) [inline telemetry/telemetrybuffer.go:343]: can you also place defer under this if..so no need to check if windows in 2 places

**Review by tamilmani1989 — CHANGES_REQUESTED** [review]

File: telemetry/telemetrybuffer.go [hunk 2]
```diff
@@ -328,32 +328,42 @@ func (tb *TelemetryBuffer) ConnectCNIToTelemetryService(telemetryNumRetries, tel
...
-					log.Logf("lock acquire error: %v", err)
-					return errors.Wrap(err, "lock acquire error")
-				}
-			}
-			if err = tb.Cleanup(FdName); err != nil {
-				return errors.Wrap(err, "cleanup failed")
-			}
-			if err = StartTelemetryService(path, args); err != nil {
-				return errors.Wrap(err, "StartTelemetryService failed")
+		if err := tb.ConnectCNIToTelemetryServiceAttempt(telemetryNumRetries, telemetryWaitTimeInMilliseconds, netPlugin, path, args); err != nil {
```
Comment (tamilmani1989) [inline telemetry/telemetrybuffer.go:331]: can we rename this as `startAndConnectTelemetryService`

**Review by tamilmani1989 — CHANGES_REQUESTED** [review]

File: telemetry/telemetrybuffer.go (hunk 2 above)
Comment (AzureAhai) [reply telemetry/telemetrybuffer.go]: addressed in the new commit.

File: telemetry/telemetrybuffer.go (hunk 1 above)
Comment (AzureAhai) [reply telemetry/telemetrybuffer.go:343]: addressed in the new commit

**Review by tamilmani1989 — APPROVED** [review]
lgtm

**Review by tamilmani1989 — APPROVED** [review]

**Pull request merged**